build_tools/fetch_sources.py

```diff
@@ -6,14 +6,37 @@
 import argparse
 from pathlib import Path
 import shlex
+import shutil
 import subprocess
 import sys
+import urllib.request
 
-THEROCK_DIR = Path(__file__).resolve().parent.parent
+THIS_SCRIPT_DIR = Path(__file__).resolve().parent
+THEROCK_DIR = THIS_SCRIPT_DIR.parent
 DEFAULT_SOURCES_DIR = THEROCK_DIR / "sources"
 PATCHES_DIR = THEROCK_DIR / "patches"
 
 
+def setup_repo_tool() -> Path:
+    """Sets up https://gerrit.googlesource.com/git-repo/, downloading as needed."""
+
+    repo_path = shutil.which("repo")
+    if repo_path:
+        print(f"Found 'repo' on PATH at '{repo_path}', using it")
+        return repo_path
+
+    repo_path = THIS_SCRIPT_DIR / "repo"
+    if repo_path.exists():
+        print(f"Found 'repo' in script dir at '{repo_path}', using it")
+        return repo_path
+
+    print(f"Unable to find 'repo', downloading into script dir at '{repo_path}'")
+    urllib.request.urlretrieve(
+        "https://storage.googleapis.com/git-repo-downloads/repo", repo_path
+    )
```

Comment:
Ah, probably want to run the equivalent of `chmod a+rx ~/.bin/repo` here so Linux users can run it standalone too.

Reply:
`os.chmod(repo_path, 0744)`

Should no-op on windows.